### PR TITLE
Copy architecture description to Sphinx docs

### PR DESCRIPTION
**Pull request by Devjiu**
This commit copies and adopts md code to rst format.

Based on pr #418.

### CODE REVIEW

**Review by kurapov-peter — APPROVED** [review]
LGTM

File: docs/getting-started/architecture.rst [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+
+Components interaction surface is limited to synchronization, that is, passing and consuming events and waiting on them. Having one component use low-level runtime and the other one a high-level creates interoperability issue. This is partially covered by existing functionality such as `Native Driver Access <https://oneapi-src.github.io/unified-runtime/core/PROG.html#native-driver-access>`_ but there are scenarios in which the support is not enough. An example of this would be a component using L0 runtime submitting a kernel and passing control flow to a component that uses SYCL runtime. The latter does not wait on the native L0 queue to drain before reading the data.
```
Comment (avladimi) [inline docs/getting-started/architecture.rst:598]: Having one component use a low-level runtime while the other uses a high-level one creates interoperability issues.

File: docs/getting-started/architecture.rst [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
+
+Pytorch has a `mechanism of streams <https://pytorch.org/docs/stable/generated/torch.cuda.Stream.html#torch.cuda.Stream>`_ (a linear sequence of execution that belongs to a specific device) to orchestrate kernel invocation and data movement. Using PyTorch’s synchronization abstractions can help decouple Triton kernels and Aten operator implementations by communicating via a stream (handled at the Inductor level). The stream can use a low-level runtime and have wrappers for SYCL-based consumers. Its implementation will live inside PyTorch as a separate component.
+
+There is also an option to introduce SYCL kernel invocation support into Unified runtime to make it indifferent to the input. This path allows for putting Unified runtime as the first-class citizen tool for all the components.
```
Comment (avladimi) [inline docs/getting-started/architecture.rst:610]: Rewrite to: Another option is to integrate SYCL kernel invocation support into the Unified runtime, making it agnostic to the input. This approach positions the Unified runtime as a versatile tool for all components.

This enhancement enables Triton to operate independently of PyTorch, eliminating unnecessary dependencies on SYCL runtime, while still leveraging the beneficial attributes of a low-level runtime.

**Review by avladimi — COMMENTED** [review]
Technical Writing review

**Pull request merged**